src/vm/jitinterface.cpp

```diff
@@ -1945,6 +1945,53 @@ CEEInfo::getClassSize(
     return result;
 }
 
+unsigned
+CEEInfo::getHeapClassSize(
+    CORINFO_CLASS_HANDLE clsHnd)
+{
+    CONTRACTL{
+        SO_TOLERANT;
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    unsigned result = 0;
+
+    JIT_TO_EE_TRANSITION_LEAF();
+
+    TypeHandle VMClsHnd(clsHnd);
+    MethodTable* pMT = VMClsHnd.GetMethodTable();
+    _ASSERTE(pMT);
+    _ASSERTE(!pMT->IsValueType());
+    result = pMT->GetBaseSize();
```

Comment:
I think this should return `GetNumInstanceFieldBytes()`. `GetBaseSize` has extra padding for minimum object size. You should not need this padding for stack allocated objects.

Reply:
I changed this to GetNumInstanceFieldBytes() + OBJECT_SIZE to account for method table pointer.